clients/admin-ui/src/features/connection-type/connection-type.slice.ts

```diff
@@ -112,9 +120,54 @@ export const connectionTypeApi = baseApi.injectEndpoints({
       Page_ConnectionSystemTypeMap_,
       Partial<ConnectionTypeParams>
     >({
-      query: (filters) => ({
-        url: `${CONNECTION_TYPE_ROUTE}${mapFiltersToSearchParams(filters)}`,
-      }),
+      async queryFn(filters, api, extraOptions, baseQuery) {
+        const firstPage = await baseQuery({
+          url: `${CONNECTION_TYPE_ROUTE}${mapFiltersToSearchParams({ ...filters, page: 1, size: 100 })}`,
+        });
+
+        if (firstPage.error) {
+          return { error: firstPage.error };
+        }
+
+        const data = firstPage.data as Page_ConnectionSystemTypeMap_;
+        const totalPages = data.pages ?? 1;
+
+        if (totalPages <= 1) {
+          return { data };
+        }
+
+        // Fetch remaining pages in parallel
+        const remainingPages = await Promise.all(
+          Array.from({ length: totalPages - 1 }, (_, i) =>
+            baseQuery({
+              url: `${CONNECTION_TYPE_ROUTE}${mapFiltersToSearchParams({
+                ...filters,
+                page: i + 2,
+                size: 100,
```

Comment:
Should probably assign `100` to a `const` and reuse it in both places.

Also I understand why we use `2` here but I'd like to see it in a variable that explains that in the name so I can shut my brain off while reading.